orttraining/orttraining/python/training/ortmodule/_training_manager.py

```diff
@@ -39,12 +38,16 @@ def execution_session_run_forward(execution_session, onnx_model, gradient_accumu
         forward_inputs = C.OrtValueVector()
         forward_inputs.reserve(len(inputs))
         for input in inputs:
-            forward_inputs.push_back(to_dlpack(input), input.dtype == torch.bool)
+            forward_inputs.push_back(_utils._torch_tensor_to_dlpack(input), input.dtype == torch.bool)
 
         forward_outputs = C.OrtValueVector()
         # Run and return module outputs.
         execution_session.run_forward(forward_inputs, forward_outputs, state, gradient_accumulation_manager.cache)
-        user_outputs = gradient_accumulation_manager.extract_outputs_and_maybe_update_cache(forward_outputs)
+        device = _utils.get_device_from_inputs(inputs, None)
```

Comment:
Instead of extracting the device again, may we pass it into this function as an argument?

Reply:
i just want to avoid the risk of function signature change. the cost of extract from inputs shouldn't be high.

Reply:
I think it is ok to change the signature of the function. We already pass in the `device` as input to the `execution_session_run_forward` in the `_inference_manager.py`. Should be ok to do it here as well. Will also look cleaner.

Reply:
ok, update the implementation.